### PR TITLE
More expected error messages added

### PR DESCRIPTION
**Pull request by ZuleykhaPavlichenkova-TomTom**
This PR adds some more expected error messages to the test cases with the empty expected result for statement error according to the https://github.com/duckdblabs/duckdb-internal/issues/2053#event-12874954527.
Changed test cases:
incomplete_checkpoint.test
vacuum_table_with_generated_column.test
in_clause_optimization_error.test
returning_clause_sizes_not_match.test
missing_subquery_error.test
utf8_invalid_unicode.test
overflow_varchar_decimal_cast.test
index_on_altered_table.test
hash_finalize_race.test
message to alter_table_rowid.test
floating_point_conversion_overflow.test

### CODE REVIEW

File: test/fuzzer/pedro/incomplete_checkpoint.test [hunk 1]
```diff
@@ -30,15 +30,15 @@ PRAGMA DEBUG_CHECKPOINT_ABORT = 'after_free_list_write';
...
 statement error
 CHECKPOINT;
 ----
+<REGEX>:FATAL Error.*Checkpoint aborted.*because of.*checkpoint_abort flag.*
 
-# error Checkpoint aborted after free list write because of PRAGMA checkpoint_abort flag
 statement error
 CHECKPOINT;
 ----
+<REGEX>:FATAL Error.*Checkpoint aborted.*because of.*checkpoint_abort flag.*
```
Comment (Tmonster) [inline test/fuzzer/pedro/incomplete_checkpoint.test:41]: here too

File: test/fuzzer/pedro/utf8_invalid_unicode.test [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 # group: [pedro]
 
-# invalid utf8, unicode value too high
+# unicode value too high
 statement error
 SELECT upper(decode('\xF6\x96\xB0\x85'::BLOB)::VARCHAR);
 ----
+<REGEX>:Conversion Error.*Failure in decode.*blob to UTF8.*
 
-# invalid utf8, unicode value too high
```
Comment (Tmonster) [inline test/fuzzer/pedro/utf8_invalid_unicode.test:9]: the comments here can stay as well

**Review by Tmonster — APPROVED** [review]
It looks good, but some of the error messages seem a little bit wordy. I'm not sure if it's necessary to have a check on something like 
`<REGEX>:FATAL Error.*Checkpoint aborted.*because of.*checkpoint_abort flag.*`
as the expected error message. 

Pinging others to let me know if they might have ideas on shortening the expected error messages

**Review by taniabogatsch — APPROVED** [review]

**Comment by taniabogatsch** [comment]
> Pinging others to let me know if they might have ideas on shortening the expected error messages

Except for the `.*` in the sentences, I don't mind them. I can understand the verbose-message argument, but at least the messages in this PR seemed readable to me—no strong opinions on this from my side.

File: test/fuzzer/pedro/incomplete_checkpoint.test [hunk 3]
```diff
@@ -30,15 +30,15 @@ PRAGMA DEBUG_CHECKPOINT_ABORT = 'after_free_list_write';
 statement ok
 CREATE SCHEMA s3;
 
-# error Checkpoint aborted after free list write because of PRAGMA checkpoint_abort flag
 statement error
 CHECKPOINT;
 ----
+<REGEX>:FATAL Error.*Checkpoint aborted.*because of.*checkpoint_abort flag.*
```
[the remark this answers is not in the document]
Comment (hmeriann) [reply test/fuzzer/pedro/incomplete_checkpoint.test:36]: This one is repeating the same thing twice

File: test/fuzzer/pedro/utf8_invalid_unicode.test (hunk 2 above)
Comment (hmeriann) [reply test/fuzzer/pedro/utf8_invalid_unicode.test:9]: returned all these comments. Should I remove the comments repeating the error messages?

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**